### ISSUE TITLE
Cannot fetch HTTPS using curl table when using the `--tls_server_certs` flag for TLS pinning

### ISSUE THREAD

**Issue by bradleyjkemp** [osquery/osquery#6964]
# Bug report

We use the `--tls_server_certs` flag to do certificate pinning on the connection between OSQuery clients and our backend server (config, logging, + distributed queries).
We've followed [this documentation page](https://osquery.readthedocs.io/en/stable/deployment/remote/#remote-authentication).

However, because the `curl` table [re-uses the same TLS transport](https://github.com/osquery/osquery/blob/master/osquery/tables/networking/curl.cpp#L32), all HTTPS requests fail because they're trying to verify certificates using the same certificate chain as for our backend server.

### What operating system and version are you using?

 version = 10.15.7
   build = 19H524
platform = darwin

### What version of osquery are you using?

version = 4.5.1

### What steps did you take to reproduce the issue?

1. Create a certificate pin for example.com e.g. using the command
```
openssl s_client -showcerts -connect example.com:443 -servername example.com.com  </dev/null | sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p' > example.com.pem
```
2. Start osquery using this pin: `osqueryi --tls_server_certs example.com.pem`
3. Run `select * from curl where url = "https://example.com";` and observe the HTTPS request succeeds
4. Run `select * from curl where url = "https://github.com";` and observe the error `W0223 16:52:43.903846 373157312 curl.cpp:83] Error making request: certificate verify failed`
 
### What did you expect to see?

You should be able to pin the certificate used for TLS communication with an OSQuery backend without affecting the ability to use the `curl` table to fetch sites that don't match this certificate chain.

I expected `--tls_server_certs` to only affect communication with the OSQuery backend and the `curl` table to still use the OS's certificate bundle.

### What did you see instead?

Pinning the TLS cert used to communicate with the OSQuery backend also affects the certificates accepted by the `curl` table.

**Comment by theopolis** [comment]
I can confirm this is the how the implementation works. A potential solution is to implement two certificate stores or search for an OS-managed bundle that will be used by the `curl` table.

**Comment by bradleyjkemp** [comment]
Ah interesting, my assumption was that the default behaviour of the `curl` table (i.e. if the `--tls_server_certs` flag isn't set) was to trust some default/OS certificate store.

But, just checked and actually the `curl` table always uses exactly the same root certs as the TLS remote plugin. So this behaviour is less unexpected than I thought 🙂 

Allowing the `curl` table to search for an OS-managed bundle would be my preferred outcome (i.e. not have to manage my own root certificates bundle) but any solution that allows two separate certificate stores would work for me.
This doesn't really block me in any way though, just confused me when I first ran into it.

**Comment by theopolis** [comment]
I agree that using the OS store would be preferred. The challenge here is that this is quite a bit of work. I'd support a PR that added this change both for the `curl` table and any TLS usage within osquery. The work is challenging because our TLS implementation is OpenSSL based so you'll have to bridge the OS cert store with the OpenSSL APIs (retrieve cert data, pass to OpenSSL API). And you'll have to do this for macOS and Windows, Linux will be the most straightforward but I am not 100% up2date if different distributions are using different paths for their OpenSSL cert bundles.

Alternatively we can clean up (make clearer) the documentation for the [table](https://github.com/osquery/osquery/blob/master/specs/curl.table) and/or the [flag](https://github.com/osquery/osquery/blob/master/docs/wiki/installation/cli-flags.md). Is this something you could submit a PR for?

**Comment by directionless** [comment]
> The work is challenging because our TLS implementation is OpenSSL based so you'll have to bridge the OS cert store with the OpenSSL APIs

Do we already do this in the `certificates` table? Or is that not refactorable here?

**Comment by lucasmrod** [comment]
Hi folks!

Out of curiosity, why isn't the `curl` table implementation using libcurl to perform the requests? (I would expect that given the name of the table..., but I'm probably missing something.)
